src/containers/ActivityContainer/FilterModal.tsx

```diff
@@ -0,0 +1,254 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Image from 'next/image';
+
+import CButton from 'src/components/CButton';
+import { useAppSelector } from 'src/hooks/useRedux';
+import { IToken } from 'src/reducers/tokens';
+
+import close from 'public/images/whiteClose.svg';
+import blackClose from 'public/images/close.svg';
+import searchLogo from 'public/images/search.svg';
+import { IFilterTokens } from 'src/constants/types';
+import defaultToken from 'public/images/defaultToken.svg';
+import CBottomSheet from 'src/components/CBottomSheet';
+
+type ModalProps = {
+  open: boolean;
+  closeModal: () => void;
+  initialSentChecked: boolean;
+  selectedTokenValue: IToken[];
+  setIsOpen: (_: boolean) => void;
+  initialReceivedChecked: boolean;
+  setSelectedTokenValue: React.Dispatch<React.SetStateAction<IToken[]>>;
+  handleSubmitFilter: (filteredValuesObject: IFilterTokens, selectedTokens: IToken[]) => void;
+};
+
+const FilterModal = ({
+  open,
+  setIsOpen,
+  closeModal,
+  handleSubmitFilter,
+  selectedTokenValue,
+  setSelectedTokenValue,
+  initialReceivedChecked,
+  initialSentChecked,
+}: ModalProps) => {
+  const [inputValue, setInputValue] = useState('');
+  const [isListVisible, setIsListVisible] = useState(false);
+  const [receivedChecked, setReceivedChecked] = useState(true);
+  const [sentChecked, setSentChecked] = useState(true);
+
+  const tokens = useAppSelector((store) => store.tokens);
+  const modalRef = useRef<HTMLDivElement | null>(null);
+  const secondModalRef = useRef<HTMLUListElement | null>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (modalRef.current && !modalRef.current.contains(event.target as Node)) {
+        closeModal();
+      }
+    };
+
+    if (open) {
+      document.addEventListener('mousedown', handleClickOutside);
+    } else {
+      document.removeEventListener('mousedown', handleClickOutside);
+    }
+    setReceivedChecked(initialReceivedChecked);
+
+    setSentChecked(initialSentChecked);
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
```

Comment:
What's this? can you explain it?

Reply:
`handleClickOutside` function checks if a user clicks outside the modal and closes it if they do.
When the modal is open, we listen for clicks outside the modal (`mousedown` event). If the modal is closed, we stop listening for those clicks.
Every time this effect runs, it resets the checked states to their initial values.

So, in essence, this `useEffect` hook listens for clicks outside the modal to close it.

Reply:
I don't think this is the best place to put it. I think the ideal scenario would be that the Modal component should implement this by default so that we wouldn't need to write it in multiple places (ConnectWallet, and here). See if you can do that and remove this logic from the component.

Reply:
You could create a custom hook that takes the ref by default, it will make the code more readable, but the problem still remains (using it in multiple places instead of modal component itself).